File: daily_challenge_leetcode/1625.js
/**
 * @param {string} s
 * @param {number} a
 * @param {number} b
 * @return {string}
 */
var findLexSmallestString = function(s, a, b) {
    let smallestString = s;
    const visited = new Set();
    const queue = [];
    queue.push(s);
    visited.add(s);
    
    while (queue.length > 0) {
        const curr = queue.shift();
        if (curr < smallestString) {
            smallestString = curr;
        }
        
        let tempArr = curr.split("");
        for (let i = 1; i < tempArr.length; i += 2) {
            tempArr[i] = ((parseInt(tempArr[i], 10) + a) % 10).toString();
        }
        const added = tempArr.join("");
        if (!visited.has(added)) {
            visited.add(added);
            queue.push(added);
        }

        const rotated = curr.slice(-b) + curr.slice(0, curr.length - b);
        if (!visited.has(rotated)) {
            visited.add(rotated);
            queue.push(rotated);
        }
    }
    
    return smallestString;
};

File: daily_challenge_leetcode/1625.py
class Solution:
    def rotate(self, s: str, b: int) -> str:
        s = s[::-1]
        s = s[:b][::-1] + s[b:][::-1]
        return s
    def findLexSmallestString(self, s: str, a: int, b: int) -> str:
        smallestString = s
        visited = set()
        que = deque([s])
        visited.add(s)

        while que:
            curr = que.popleft()
            if curr < smallestString:
                smallestString = curr
            temp = list(curr)
            for i in range(1, len(temp), 2):
                temp[i] = str((int(temp[i]) + a) % 10)
            temp = "".join(temp)

            if temp not in visited:
                visited.add(temp)
                que.append(temp)

            rotated = self.rotate(curr, b)
            if rotated not in visited:
                visited.add(rotated)
                que.append(rotated)
        return smallestString
